packages/http-server/src/http-server.ts

```diff
@@ -35,42 +56,60 @@ export class HttpServer {
   private _listening: boolean = false;
   private _protocol: HttpProtocol;
   private _address: AddressInfo;
-  private httpRequestListener: HttpRequestListener;
-  private httpServer: Server;
+  private _httpsOptions: HttpsOptions = {};
+  private requestListener: RequestListener;
+  private server: Server | HttpsServer;
 
   /**
-   * @param httpServerOptions
-   * @param httpRequestListener
+   * @param requestListener
+   * @param serverOptions
    */
-  constructor(
-    httpRequestListener: HttpRequestListener,
-    httpServerOptions?: HttpServerOptions,
-  ) {
-    this.httpRequestListener = httpRequestListener;
-    if (!httpServerOptions) httpServerOptions = {};
-    this._port = httpServerOptions.port || 0;
-    this._host = httpServerOptions.host || undefined;
-    this._protocol = httpServerOptions.protocol || 'http';
+  constructor(requestListener: RequestListener, serverOptions?: ServerOptions) {
+    this.requestListener = requestListener;
+    if (!serverOptions) serverOptions = {};
+    this._port = serverOptions.port || 0;
+    this._host = serverOptions.host || undefined;
+    this._protocol = serverOptions.protocol || 'http';
+    if (this._protocol === 'https') {
+      this._httpsOptions.cert = serverOptions.cert;
+      this._httpsOptions.key = serverOptions.key;
+    }
   }
 
   /**
    * Starts the HTTP / HTTPS server
    */
   public async start() {
-    this.httpServer = createServer(this.httpRequestListener);
-    this.httpServer.listen(this._port, this._host);
-    await pEvent(this.httpServer, 'listening');
-    this._listening = true;
-    this._address = this.httpServer.address() as AddressInfo;
+    const options: CreateServerOptions = {};
+    if (this._protocol === 'https') {
+      if (this._httpsOptions.key) {
+        options.key = await readFile(this._httpsOptions.key);
```

Comment:
It's not always the case that certificates are stored in files. For example, https://12factor.net recommends to pass all configuration via environment variables.

IMO, it's not the responsibility of `HttpServer` to figure out how to obtain key/cert/pfx - that's up to the user (caller).

Please remove `readFile` calls and make `HttpServer` API closer to what Node.js core provides, where the users must pass Buffer/String values for key/cert/pfx/etc.